### PR TITLE
[DLFL22] Clarify data "dependencies" in 09-echo_data.

### PR DESCRIPTION
**Pull request by pscollins**
* Add comments to explain that successive batches are not independent.
* Reset hidden state whenever EchoData state is reset.
* Expand the final "example" section to make it easier to read out when the model is correct.
* Use a fixed numpy seed for reproducibility.

### CODE REVIEW

File: 09-echo_data.ipynb [hunk 1]
```diff
@@ -18,10 +18,11 @@
     "from res.sequential_tasks import EchoData\n",
```
Comment (Atcold) [inline 09-echo_data.ipynb:18]: <pre class="ql-syntax" spellcheck="false">Line #15.    #   [[S11 S12...S1N], [S22 S22...S2N], ..., [SM1 SM2...SMN]]
</pre><p>The first <code>S22</code> should be <code>S21</code> instead.  </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Atcold/pytorch-Deep-Learning/pull/830/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (Atcold) [inline 09-echo_data.ipynb:18]: <pre class="ql-syntax" spellcheck="false">Line #19.            # we share the hidden state across successive invocations of .forward.
</pre><p>You shouldn't talk about calling the <code>.forward()</code> method. We are actually calling the model itself, which does additional stuff. If you're looking for the generic term of forward propagation, just type that in plain English. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Atcold/pytorch-Deep-Learning/pull/830/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

Comment (Atcold) [inline 09-echo_data.ipynb:18]: <pre class="ql-syntax" spellcheck="false">Line #5.        hidden = None   
</pre><p>Ah, I see. Not sure why <code>hidden</code> was zeroed outside… </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Atcold/pytorch-Deep-Learning/pull/830/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>

Comment (Atcold) [inline 09-echo_data.ipynb:18]: <pre class="ql-syntax" spellcheck="false">Line #1.    import numpy as np
</pre><p>You've already imported <code>numpy</code>…</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Atcold/pytorch-Deep-Learning/pull/830/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

**Comment by pscollins** [comment]
(My git is a bit rusty -- not sure if the convention is to squash the fixes into the original commit or add a new one, hope that squashing was the right thing to do)

File: 09-echo_data.ipynb (hunk 1 above)
Comment (pscollins) [reply 09-echo_data.ipynb:18]: Deleted

Comment (pscollins) [reply 09-echo_data.ipynb:18]: Thanks! `Changed` it to just say `successive invocations`.

Comment (pscollins) [reply 09-echo_data.ipynb:18]: Thanks! Done.

**Comment by Atcold** [comment]
> (My git is a bit rusty -- not sure if the convention is to squash the fixes into the original commit or add a new one, hope that squashing was the right thing to do)

I do the squashing when everything looks alright 😉
